ibtk/src/lagrangian/fortran/lagrangian_interaction2d.f.m4

```diff
@@ -712,72 +737,71 @@ c
 c
 c     Local variables.
 c
+      INTEGER ilower(0:NDIM-1),iupper(0:NDIM-1)
       INTEGER ic0,ic1
       INTEGER ic_center(0:NDIM-1),ic_lower(0:NDIM-1),ic_upper(0:NDIM-1)
-      INTEGER d,l,s
+      INTEGER d,l,s,nugc(0:NDIM-1)
 
       REAL X_cell(0:NDIM-1),w(0:NDIM-1,0:3)
 c
 c     Prevent compiler warning about unused variables.
 c
       x_upper(0) = x_upper(0)
 c
+c     Setup convenience arrays.
+c
+      ilower(0) = ilower0
+      ilower(1) = ilower1
+
+      iupper(0) = iupper0
+      iupper(1) = iupper1
+
+      nugc(0) = nugc0
+      nugc(1) = nugc1
+c
 c     Use the piecewise cubic delta function to interpolate u onto V.
 c
       do l = 0,nindices-1
          s = indices(l)
+         do d=0,NDIM-1
 c
 c     Determine the Cartesian cell in which X(s) is located.
 c
-         ic_center(0) =
-     &        floor((X(0,s)+Xshift(0,l)-x_lower(0))/dx(0))
-     &        + ilower0
-         ic_center(1) =
-     &        floor((X(1,s)+Xshift(1,l)-x_lower(1))/dx(1))
-     &        + ilower1
-
-         X_cell(0) = x_lower(0)+(dble(ic_center(0)-ilower0)+0.5d0)*dx(0)
-         X_cell(1) = x_lower(1)+(dble(ic_center(1)-ilower1)+0.5d0)*dx(1)
+            ic_center(d) =
+     &           floor((X(d,s)+Xshift(d,l)-x_lower(d))/dx(d))
+     &           + ilower(d)
+            X_cell(d) = x_lower(d)
+     &           +(dble(ic_center(d)-ilower(d))+0.5d0)*dx(d)
 c
 c     Determine the interpolation stencil corresponding to the position
 c     of X(s) within the cell.
 c
-         do d = 0,NDIM-1
             if ( X(d,s).lt.X_cell(d) ) then
                ic_lower(d) = ic_center(d)-2
                ic_upper(d) = ic_center(d)+1
             else
                ic_lower(d) = ic_center(d)-1
                ic_upper(d) = ic_center(d)+2
             endif
-         enddo
 
-         ic_lower(0) = max(ic_lower(0),ilower0-nugc0)
-         ic_upper(0) = min(ic_upper(0),iupper0+nugc0)
-
-         ic_lower(1) = max(ic_lower(1),ilower1-nugc1)
-         ic_upper(1) = min(ic_upper(1),iupper1+nugc1)
+            ic_lower(d) = max(ic_lower(d),ilower(d)-nugc(d))
+            ic_upper(d) = min(ic_upper(d),iupper(d)+nugc(d))
 c
 c     Compute the interpolation weights.
 c
-         do ic0 = ic_lower(0),ic_upper(0)
-            X_cell(0) = x_lower(0)+(dble(ic0-ilower0)+0.5d0)*dx(0)
-            w(0,ic0-ic_lower(0)) =
-     &           lagrangian_piecewise_cubic_delta(
-     &           (X(0,s)+Xshift(0,l)-X_cell(0))/dx(0))
-         enddo
-
-         do ic1 = ic_lower(1),ic_upper(1)
-            X_cell(1) = x_lower(1)+(dble(ic1-ilower1)+0.5d0)*dx(1)
-            w(1,ic1-ic_lower(1)) =
-     &           lagrangian_piecewise_cubic_delta(
-     &           (X(1,s)+Xshift(1,l)-X_cell(1))/dx(1))
+            do ic0 = ic_lower(d),ic_upper(d)
```

Comment:
Could we change `ic0` to `ic` here? I think that I pretty consistently use `ic0` to mean a "cell index in the 0 axis".